File: Taller de Haskell/Problema 9/problema9.py
# Pedido de la pizza
print("¿Quieres una pizza vegetariana? (sí/no)")
respuesta = input().lower()

if respuesta == "sí" or respuesta == "si":
    es_veg = True
    ingredientes_base = ["mozzarella", "tomate"]
    opciones = ["pimiento", "tofu"]
else:
    es_veg = False
    ingredientes_base = ["mozzarella", "tomate"]
    opciones = ["peperoni", "jamón", "salchicha"]

print("Elige un ingrediente adicional:")
for i, ingr in enumerate(opciones):
    print(f"{i + 1}. {ingr}")

eleccion = int(input("Ingrese el número de su elección: "))
ingrediente_adicional = opciones[eleccion - 1]

print("\n--- Pedido final ---")
print("Tipo de pizza:", "Vegetariana" if es_veg else "No vegetariana")
print("Ingredientes: ", ", ".join(ingredientes_base + [ingrediente_adicional]))

File: Taller de Haskell/Problema 9/problema9.hs
import Data.Char (toLower)

--Pedido
descripcionPizza :: Bool -> String -> (String, Bool)
descripcionPizza esVegetariana ingrediente
    | esVegetariana = ("Pizza vegetariana con mozzarella, tomate y " ++ ingrediente, True)
    | otherwise = ("Pizza no vegetariana con mozzarella, tomate y " ++ ingrediente, False)

main :: IO ()
main = do
    putStrLn "¿Quieres una pizza vegetariana? (si/no)"
    respuesta <- getLine
    let esVegetariana = map toLower respuesta `elem` ["si", "sí"]

    let opciones = if esVegetariana then ["pimiento", "cebolla"]
                                    else ["peperoni", "jamón", "salchicha"]

    putStrLn "Elige un ingrediente adicional: "
    mapM_ putStrLn [show i ++ ". " ++ ingr | (i, ingr) <- zip [1..] opciones]

    opcion <- getLine
    let indice = read opcion :: Int
    let ingrediente = opciones !! (indice - 1)

    let (descripcion, _) = descripcionPizza esVegetariana ingrediente

    putStrLn "\nPedido:"
    putStrLn descripcion
